napari/layers/utils/layer_utils.py

```diff
@@ -684,58 +684,201 @@ def features_to_pandas_dataframe(features: Any) -> pd.DataFrame:
+                for name, column in self._values.items()
+            },
+            index=range(1),
+            copy=True,
         )
-    else:
-        features = _validate_features(features, num_data=num_data)
-    defaults = pd.DataFrame(
-        {
-            name: _get_default_column(column)
-            for name, column in features.items()
-        },
-        index=range(1),
-        copy=True,
-    )
-    return features, defaults
+
+    @property
+    def defaults(self) -> pd.DataFrame:
+        """The default values one-row table."""
+        return self._defaults
+
+    def properties(self) -> Dict[str, np.ndarray]:
+        """Converts this to a deprecated properties dictionary.
+
+        This will reference the features data when possible, but in general the
+        returned dictionary mary contain copies of those data.
+
+        Returns
+        -------
+        Dict[str, np.ndarray]
+            The properties dictionary equivalent to the given features.
+        """
+        return _features_to_properties(self._values)
+
+    def choices(self) -> Dict[str, np.ndarray]:
+        """Converts this to a deprecated property choices dictionary.
+
+        Only categorical features will have corresponding entries in the dictionary.
+
+        Returns
+        -------
+        Dict[str, np.ndarray]
+            The property choices dictionary equivalent to this.
+        """
+        return {
+            name: series.dtype.categories.to_numpy()
+            for name, series in self._values.items()
+            if isinstance(series.dtype, pd.CategoricalDtype)
+        }
+
+    def currents(self) -> Dict[str, np.ndarray]:
+        """Converts the defaults table to a deprecated current properties dictionary."""
+        return _features_to_properties(self._defaults)
+
+    def set_currents(
+        self,
+        currents: Dict[str, np.ndarray],
+        *,
+        update_indices: Optional[List[int]] = None,
+    ):
+        """Sets the default values using the deprecated current properties dictionary.
+
+        May also update some of the feature values to be equal to the new default values.
+        """
+        currents = coerce_current_properties(currents)
+        self._defaults = _validate_features(currents, num_data=1)
+        if update_indices is not None:
+            for k in self._defaults:
+                self._values[k][update_indices] = self._defaults[k][0]
+
+    def resize(
+        self,
+        size: int,
+    ):
+        """Resize this padding with default values if required.
+
+        Parameters
+        ----------
+        size : int
+            The new size (number of rows) of the features table.
+        """
+        current_size = self._values.shape[0]
+        if size < current_size:
+            self.remove(range(size, current_size))
+        elif size > current_size:
+            to_append = pd.concat([self._defaults] * (size - current_size))
+            self.append(to_append)
```

Comment:
Would there be any reason to warn if resize is no-op as it's already the current size ?

Reply:
This is a private implementation class, so I think it's fine not to warn. If we warn, then the callers of this method will have to check that condition.